src/openvic-simulation/types/CowPtr.hpp

```diff
@@ -0,0 +1,145 @@
+#pragma once
+
+#include <atomic>
+#include <concepts>
+#include <cstddef>
+#include <type_traits>
+#include <utility>
+
+#include "openvic-simulation/utility/Utility.hpp"
+
+namespace OpenVic {
+	template<typename T>
+	class cow_ptr {
```

Comment:
```suggestion
	class cow_ptr { //copy-on-write pointer
```
Idk if cow is a common term for cpp programmers.
For the rest of the world, it refers to the animal.